hedera-node/src/main/java/com/hedera/services/state/EntityCreator.java

```diff
@@ -20,13 +20,69 @@
  * ‍
  */
 
+import com.google.protobuf.ByteString;
 import com.hedera.services.records.RecordCache;
 import com.hedera.services.state.submerkle.ExpirableTxnRecord;
+import com.hedera.services.utils.TxnAccessor;
 import com.hederahashgraph.api.proto.java.AccountID;
-import com.hederahashgraph.api.proto.java.TransactionRecord;
+import com.hederahashgraph.api.proto.java.Timestamp;
+import com.hederahashgraph.api.proto.java.TransactionReceipt;
+
+import java.time.Instant;
 
 public interface EntityCreator {
+	/**
+	 * setter for {@link RecordCache} in {@link EntityCreator}
+	 *
+	 * @param recordCache
+	 * 		record cache
+	 */
 	void setRecordCache(RecordCache recordCache);
 
-	ExpirableTxnRecord createExpiringRecord(AccountID id, TransactionRecord record, long now, long submittingMember);
+	/**
+	 * Sets needed properties like expiry and submitting member to {@link ExpirableTxnRecord} and adds record to state
+	 * based on {@code GlobalDynamicProperties.cacheRecordsTtl}. If not it is added to be tracked for Expiry to {@link
+	 * RecordCache}
+	 *
+	 * @param id
+	 * 		account id
+	 * @param record
```

Comment:
I think javadoc isn't happy about this line since we've changed the param name from `record -->> expiringRecord`.

Reply:
Ah yes , fixed. Thanks Michael !